### DISCUSSION TITLE
Too many redirection based on the default locale

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#41777] [Help]
I'm using `next ^12.3.1` and `next-i18nnext ^12.0.1`

I have i18n config like this:

```
i18n: {
        defaultLocale: 'uk',
        locales: ["de", "uk", "es"],
        localeDetection: false,
    }
```
I have a locale structure like this:

```
example.com/uk
example.com/es
example.com/de
```

and in my folder structure is like this:

```
locales
  /uk
     common.json
  /es
     common.json
```

and I have a middleware logic like this:

`If the URL is example.com then redirect them to example.com/uk`

With this setup I get too many redirects error and when I log the output for a single requst it's something like this 15-20 times 

`incoming url https://example.com/, redirected to https://example.com/uk`

In order to overcome this problem I need to add an unused "en" locale like this:

````
i18n: {
        defaultLocale: 'en',
        locales: ["de", "uk", "es", "en"],
        localeDetection: false,
    }
````

Then everything is working fine.  I cannot find any logical explanation for this. Can you help me with it ?

**Comment by joggienl** [comment]
I think the information from the workaround on this page will help you: https://nextjs.org/docs/advanced-features/i18n-routing#prefixing-the-default-locale
